### PR TITLE
Move @Slynova/flydrive to local package + add support for Azure Blob Storage

### PR DESCRIPTION
**Pull request by TheAzack9**
As discussed with @rijkvanzanten on Discord does this PR move the flydrive package to a local folder and extends the package to include support for Azure blob storage. This is great for everyone living in the azure side of the cloud as it allows us to store files in the azure cloud. I have tried to update all documentation that i could find to show that azure blobs are included :)

I did strip tests from the flydrive package as i'm not sure if those are relevant for Directus or not. (I can add them back if requested) I am not sure what is relevant or not to keep exactly, so please just say if there are files i should remove or shouldn't have removed :) (I have not removed anything essential to the packages ofc)

I have not done extensive testing on this so far, but i surely will going forward as this is something i need myself ;) 

I hope it looks good 👍

### CODE REVIEW

File: package.json
Comment (rijkvanzanten) [inline package.json:196]: Change this to `@directus/drive`

File: api/src/storage.ts
Comment (rijkvanzanten) [inline api/src/storage.ts:6]: Lets name it `@directus/drive` instead

File: api/src/services/assets.ts
Comment (rijkvanzanten) [inline api/src/services/assets.ts:47]: Did we document this change in the readme?

File: api/package.json
Comment (rijkvanzanten) [inline api/package.json:75]: Use lowercase for the package name `azureblob` or `azure-blob`, update references

Comment (rijkvanzanten) [reply api/package.json:75]: We'll have to "lift" these sub-packages into the "root" packages folder:

`/packages/drive/packages/flydrive-azureBlob` -> `/packages/drive-azure-blob`

**Review by rijkvanzanten — COMMENTED** [review]
Instead of `azure-blob` (and potentially later `azure-file`), would it be possible to have a single `drive-azure`, and support both blobs & files based on a config flag?

File: packages/drive/.gitignore
Comment (rijkvanzanten) [inline packages/drive/.gitignore:1]: Lets use the same gitignore / eslint / editorconfig setup as the other packages

File: packages/drive/README.md
Comment (rijkvanzanten) [inline packages/drive/README.md:1]: We'll have to clean this up / rewrite to make sense

File: packages/drive/lerna.json
Comment (rijkvanzanten) [inline packages/drive/lerna.json:1]: We'll have to get rid of this, can't have nested lerna configurations

File: packages/drive/package.json
Comment (rijkvanzanten) [inline packages/drive/package.json:2]: Needs to go in favor of being `@directus/drive` "root"

File: packages/drive/packages/flydrive-azureBlob/tsconfig.json
Comment (rijkvanzanten) [inline packages/drive/packages/flydrive-azureBlob/tsconfig.json:1]: Use root directus tsconfig

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Don't take these points as hard to-do's @TheAzack9 🙂 I can pick up where you left off 👍

**Comment by TheAzack9** [comment]
> Instead of `azure-blob` (and potentially later `azure-file`), would it be possible to have a single `drive-azure`, and support both blobs & files based on a config flag?

This should be fine to do (and probably is the best solution tbh, easy to get tunnel visioned ;) )

> Don't take these points as hard to-do's @TheAzack9 🙂 I can pick up where you left off 👍

No worries, i just want to help get the ball rolling as much as i can ;) I can gladly go through tomorrow and fix what i can. I kinda want to play around some more with it, so should be fun! :)

**Comment by TheAzack9** [comment]
Took a bit longer than planned, but i finally found the time to go through and fix the comments :)

I have tried to change stuff to be more consistent with existing configs, but there may still be some stuff i have misplaced 😅 
Short summary of changes:

- Renamed from "slynova/flydrive*" -> "directus/drive*"
- Removed learna config in sub directory. I still kept the packages in a subdirectory (/packages/drive) since it looked more organized imo and has better separation since they are related. Should be simple to move them even further if necessary. (Also allows for a general readme, which i feel is nice wrt. giving credit and such. 
- Documented the async change to getStream
- Renamed from `directus/azureBlob` to `directus/azure`
- Removed a lot of uncecessary configs that is duplicate to root

Remaining tasks:

- Update readme in packages/drive/ :)

**Comment by emregency** [comment]
Hey guys,

I was just wondering what is actually blocking this PR and how we can move it forward. I am looking forward to use Azure Blob so if there anything I can do to move this PR further let me know. It looks like great work was put in and needs just a last push.

**Comment by benhaynes** [comment]
Time. Our team has been extremely busy wrapping up other internal systems, and will _finally_ be back to the Core codebase this week. I defer to @rijkvanzanten on what can be done (if anything) to expedite.

**Comment by TheAzack9** [comment]
Just jumping in with a little ping! :) 

If there is any more i could do to help here i would love to help out 😅 I am depending a bit on this to be able to go live with a website :)

**Comment by rijkvanzanten** [comment]
@TheAzack9 Yes actually! I just recently dove into this to address #3640 and was wondering about 2 things: 

Does getStream need to be async, and

can you maybe add in support for ranges? :o

I believe every storage driver we have has native support for ranged requests, which in turn allows us to expose that on the /assets endpoint

Other than that, there's no blockers

**Comment by TheAzack9** [comment]
I thought it had to, but i see that Xalag proposed a solution that is not breaking here: https://github.com/Slynova-Org/flydrive/pull/158, so it should be possible for it to not be async :)

I can look into supporting ranges :)

**Comment by TheAzack9** [comment]
I need some more context on what you mean with ranged requests i think @rijkvanzanten :) Are you thinking of this? https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests

**Comment by rijkvanzanten** [comment]
Yes exactly! It's a way to ask for specific bytes out of a file. Browsers use this for things like video buffering, where they only requests little parts of a video at a time. 

From a quick search around, it seems like all the external libraries used support it in some way or fashion. This is how you do it in "local" (eg FS) for example:

CleanShot 2021-02-04 at 13 37 12@2x

**Pull request closed without merging**
